src/main/java/core/basesyntax/bookstore/service/impl/BookServiceImpl.java

```diff
@@ -56,21 +59,31 @@ public BookDto update(Long id, UpdateBookRequestDto requestDto) {
     }
 
     @Override
-    public Page<BookDto> findAll(BookParamsDto params, Pageable pageable) {
+    public Page<BookWithoutCategoryDto> findAll(BookParamsDto params, Pageable pageable) {
         Specification<Book> spec = bookSpecificationBuilder.build(params);
         Page<Book> page = bookRepository.findAll(spec, pageable);
-        List<BookDto> bookDtoList = page
+        List<BookWithoutCategoryDto> bookWithoutCategoryDtoList = page
                 .stream()
-                .map(bookMapper::toDto)
+                .map(bookMapper::toDtoWithoutCategory)
                 .toList();
-        return new PageImpl<>(bookDtoList, pageable, page.getTotalElements());
+        return new PageImpl<>(bookWithoutCategoryDtoList, pageable, page.getTotalElements());
     }
 
     @Override
-    public Page<BookDto> findAll(Pageable pageable) {
+    public Page<BookWithoutCategoryDto> findAll(Pageable pageable) {
         Page<Book> page = bookRepository.findAll(pageable);
-        List<BookDto> bookDtoList = page.stream()
-                .map(bookMapper::toDto)
+        List<BookWithoutCategoryDto> bookWithoutCategoryDtoList = page.stream()
+                .map(bookMapper::toDtoWithoutCategory)
+                .toList();
+        return new PageImpl<>(bookWithoutCategoryDtoList, pageable, page.getTotalElements());
+    }
+
+    @Override
+    public Page<BookWithoutCategoryDto> findAllByCategoryId(Long categoryId, Pageable pageable) {
+        Page<Book> page = bookRepository
+                .findAllByCategoriesContains(Set.of(new Category(categoryId)), pageable);
```

Comment:
why do you complicate it? Can you use `findAllByCategoryId`?

Reply:
I can, but then I'll need to add implementation for book repository